### PR TITLE
Refine distillation for Segmentation  

### PR DESCRIPTION
**Pull request by juncaipeng**
* Add  SegPairWiseLoss and SegChannelwiseLoss
* To support distributed training, detach the tensor from the eval model
* Refine the distillation framework and add some comments

### CODE REVIEW

File: paddleslim/dygraph/dist/distill.py [hunk 1]
```diff
@@ -135,42 +158,15 @@ def _get_distill_idx(self):
...
-                final_distill_dict['teacher'][feature_type + '_' + str(
-                    sidx) + '_' + str(tidx)] = self.tea_outs_dict[
-                        feature_type + '_' + str(tidx)]
-        return final_distill_dict
-
     def forward(self, *inputs, **kwargs):
         stu_batch_outs = self._student_models.forward(*inputs, **kwargs)
         tea_batch_outs = self._teacher_models.forward(*inputs, **kwargs)
+        if not self._student_models.training:
+            stu_batch_outs = [i.detach() for i in stu_batch_outs]
```
Comment (ceci3) [inline paddleslim/dygraph/dist/distill.py]: student 是不是只会有train模式的模型，不需要detach出来？

Comment (juncaipeng) [reply paddleslim/dygraph/dist/distill.py]: 是的，通常student是train模式，已经修改删掉，添加assert判断。

**Review by ceci3 — APPROVED** [review]
LGTM
